.github/workflows/tests.yml

```
@@ -0,0 +1,45 @@
+name: tests
+
+on:
+  push:
+    branches:
+      - dev
+      - main
+  pull_request:
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  tests:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: linux
+            runs-on: ubuntu-22.04
+            arch: x86_64
+
+    name: "${{ matrix.os }}-${{ matrix.arch }}"
+    runs-on: ${{ matrix.runs-on }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - name: Install Nim
+        uses: iffy/install-nim@049e1fad22d41813c77cff11ab8e1541a872014a
+        with:
+          version: "binary:2.0.0"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # Avoid rate limiting.
```

fyi the same token can be accessed via:

```suggestion
          GITHUB_TOKEN: ${{ github.token }} # Avoid rate limiting.
```

I think in CO we mostly used `github.token` but dont think github has preference to either as they both give the same value